src/zeit/content/cp/area.py

```diff
@@ -299,43 +299,104 @@ def count(self):
     @count.setter
     def count(self, value):
         self._count = value
-        self._fill_with_placeholders()
+        self.update_autopilot()
 
     def _fill_with_placeholders(self):
-        if self._automatic:
-            layouts = []
-            for key in self:
-                block = self[key]
-                if ITeaserBlock.providedBy(block):
-                    layouts.append(block.layout)
-                del self[key]
-            for i in range(self.count):
-                block = self.create_item('auto-teaser')
-                # self.count might be greater than the number of manual teasers
-                if layouts:
-                    block.layout = layouts.pop(0)
+        """Remove all blocks from the area, add automatic teaser blocks instead
+
+        Copy layout of teaser blocks to automatic teaser block in same position
+
+        To make sure we only have #count automatic teaser blocks, we call
+        update_autopilot in the end, which will add / remove automatic teaser
+        blocks accordingly.
+
+        """
+        if not self.automatic:
+            return
+
+        # Add / remove AutomaticTeaserBlocks as needed to reach #count
+        self.update_autopilot()
+
+        order = self.keys()
+        for block in self.values():
+            if not block.volatile:
+                continue
+
+            auto_block = self.create_item('auto-teaser')
+            auto_block.__name__ = block.__name__  # required to updateOrder
+
+            if ITeaserBlock.providedBy(block):
+                auto_block.layout = block.layout
+
+            # Only deletes first occurrence of __name__, i.e. `block`
+            del self[block.__name__]
+
+        # Preserve order of blocks that are kept when turning AutoPilot on.
+        self.updateOrder(order)
+
+    def update_autopilot(self):
+        """Update number of automatic teasers inside the AutoPilot.
+
+        Does not touch any block that is not an IAutomaticTeaserBlock, so only
+        the number of automatic teasers is configured via the `Area.count`
+        setting. Thus the AutoPilot may contain more than `Area.count` teasers.
+
+        """
+        if not self.automatic:
+            return
+
+        automatic_blocks = [
+            x for x in self.values() if IAutomaticTeaserBlock.providedBy(x)]
```

Comment:
Ich glaub die Frage ist was man möchte, wenn man bei einer (teilweise manuell befüllten) Fläche den Autopilot aktiviert: Sollen dann insgesamt `count` Teaser drin sein (so in etwa wäre es jetzt, richtig?) oder `count` *Auto*teaser. Ich würde es jetzt erstmal so lassen; wenn es nicht gefällt kann man's ja ändern.